.travis.yml

```diff
@@ -36,23 +36,28 @@ matrix:
       env: TOXENV=py34-alldeps-withcov-posix,codecov-publish
     - python: 3.5
       env: TOXENV=py35-alldeps-withcov-posix,codecov-publish
+    - python: 3.6
+      env: TOXENV=py36-alldeps-withcov-posix,codecov-publish
     # For now all non-trial tests are in a single job to reduce the time spent
     # on starting separate jobs.
     - python: 2.7
       env: TOXENV=narrativedocs,apidocs,pyflakes,newsfragment,manifest-checker
-    - python: 3.5
+    - python: 3.6
       env: TOXENV=pyflakes3
     # Twistedchecker is running as a separate job so that we can ignore if it
     # fails.
-    - python: 3.5
```

Comment:
twistedchecker cannot run on Python 3.6
This must remain on Python 3.5.
If you look at the logs: https://travis-ci.org/twisted/twisted/jobs/261731629 , this error is there:

```
AttributeError: 'TreeRebuilder3k' object has no attribute 'visit_joinedstr'
```

This is fixed upstream in newer versions of pylint and astroid: https://github.com/PyCQA/pylint/issues/1113

However, twistedchecker cannot use these new versions, because it monkey patches some internals
of astroid.  So we need to keep the twistedchecker stuff on Python 3.5 for now,
until we can update twistedchecker to work with newer versions of astroid.